product_subscription/models/subscription_request.py

```diff
@@ -234,8 +234,12 @@ def _validate_pending_request(self):
         for pending_request in pending_request_list:
             try:
                 pending_request.validate_request()
+                self.env.cr.commit()
             except UserError:
-                # todo: notify
+                print("Unexpected UserError")
+                continue
+            except ValidationError:
+                print("Unexpected ValidationError")
```

Comment:
It'd be cleaner to log an ERROR instead of just a print. In the log, print the error message or the stack trace. Something like 
```py
        except UserError as error:
            _logger.info(error.name)
            _logger.debug("Traceback:", exc_info=True)
```

Reply:
error.name is empty... didn't find what is containing the error message. anyway I'll use _logger instead

Reply:
did you catch the error with the `as` expression like in the example?

Reply:
Now yes and it seems to display the name in the logs while expecting the object with debug wasn't giving anything.

Reply:
I copy pasted the code above. You'd need to set the log level to debug to see the `logger.debug` statement. I'd log everything here with `_logger.error(...)`.